### PR TITLE
3.3.x support

### PR DESCRIPTION
**Pull request by battye**
This PR contains the changes so Titania can run on phpBB v3.3.1+

These include:

* Upgraded Symfony version for Translation Validator
* Some service fixes by @paul999 
* Port the old Symfony regex code to maintain backwards compatibility

### CODE REVIEW

File: controller/helper.php [hunk 1]
```diff
@@ -13,7 +13,21 @@
...
+use phpbb\db\driver\driver_interface;
+use phpbb\event\dispatcher;
+use phpbb\language\language;
+use phpbb\request\request_interface;
+use phpbb\routing\helper as routing_helper;
+use phpbb\symfony_request;
+use phpbb\template\template;
 use phpbb\titania\config\config as ext_config;
+use phpbb\user;
+use Symfony\Component\HttpFoundation\JsonResponse;
```
Comment (iMattPro) [inline controller/helper.php:29]: JsonResponse is not used

**Review by iMattPro — APPROVED** [review]
Several files are missing new lines at the end of them which need to be resolved.

**Comment by iMattPro** [comment]
Looks like we've got the circular reference bug with Titania and 3.3.1 here.

```
Circular reference detected for service "cron.manager", path: "controller.helper -> cron.manager -> phpbb.titania.cron.task.rebuild_composer_repo -> phpbb.titania.manage.tool.composer.rebuild_repo -> phpbb.titania.contribution.type.collection -> phpbb.titania.style.type -> phpbb.titania.style.demo -> phpbb.titania.controller.helper -> cron.manager". in /titania/vendor/symfony/dependency-injection/Container.php:297
Stack trace:
#0 /titania/vendor/symfony/dependency-injection/ContainerBuilder.php(606): Symfony\Component\DependencyInjection\Container->get('cron.manager', 2)
#1 /titania/vendor/symfony/dependency-injection/ContainerBuilder.php(1281): Symfony\Component\DependencyInjection\ContainerBuilder->doGet('cron.manager', 1, Array, true)
#2 /titania/vendor/symfony/dependency-injection/ContainerBuilder.php(1238): Symfony\Componen in /titania/vendor/symfony/dependency-injection/Container.php on line 297
```

**Comment by DavidIQ** [comment]
We will need to pull in the 3.3.2 fix into the .com repo, if not already done. @paul999

File: controller/helper.php (hunk 1 above)
Comment (DavidIQ) [inline controller/helper.php:29]:
```suggestion
use phpbb\user;
```

Comment (DavidIQ) [reply controller/helper.php:29]:
```suggestion
```

File: config/services.yml [hunk 2]
```diff
@@ -482,4 +492,4 @@ services:
             - '@auth'
             - '@phpbb.titania.translation.prevalidator'
         tags:
-            - { name: titania.contribution.type }
+            - { name: titania.contribution.type }
```
Comment (DavidIQ) [inline config/services.yml:495]:
```suggestion
            - { name: titania.contribution.type }

```

File: includes/library/expression/Expression.php [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+	}
+
+	/**
+	 * @return Regex
+	 */
+	public function getRegex()
+	{
+		return self::TYPE_REGEX === $this->value->getType() ? $this->value : $this->value->toRegex();
+	}
+}
```
Comment (DavidIQ) [inline includes/library/expression/Expression.php:146]:
```suggestion
}

```

File: includes/library/expression/Glob.php [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+	 *
+	 * @return Regex
+	 */
+	public function toRegex($strictLeadingDot = true, $strictWildcardSlash = true)
+	{
+		$regex = FinderGlob::toRegex($this->pattern, $strictLeadingDot, $strictWildcardSlash, '');
+
+		return new Regex($regex);
+	}
+}
```
Comment (DavidIQ) [inline includes/library/expression/Glob.php:106]:
```suggestion
}

```

File: includes/library/expression/Regex.php [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+			$pattern = substr($pattern, 0, -1);
+		}
+
+		if ($this->endJoker = (self::JOKER === substr($pattern, -2) && self::ESCAPING !== substr($pattern, -3, -2))) {
+			$pattern = substr($pattern, 0, -2);
+		}
+
+		$this->pattern = $pattern;
+	}
+}
```
Comment (DavidIQ) [inline includes/library/expression/Regex.php:319]:
```suggestion
}

```

**Pull request merged**